### PR TITLE
chore: Reference lifecycle manager api module

### LINKED ISSUE #1875: Replace LM dependency in the CLI with the the api module
**Description**

Since we introduced the lifecycle manager api as a go module, we need to reference it in the CLI instead of the whole lifecycle manager dependency

**ACs:**

- [ ] Replace lifecycle manager dependency in the CLI with the LM api

### PR DESCRIPTION
**Pull request by nesmabadr**
**Description**

Changes proposed in this pull request:

- Reference lifecycle manager api module as a dependency instead of lifecycle manager

**Related issue(s)**
Resolves #1875

### CODE REVIEW

File: tests/e2e/e2e_suite_test.go [hunk 1]
```diff
@@ -35,7 +32,6 @@ func TestE2e(t *testing.T) {
 
 var _ = BeforeSuite(func() {
 	ctx, cancel = context.WithCancel(context.TODO())
-	logf.SetLogger(log.ConfigLogger(9, zapcore.AddSync(GinkgoWriter)))
```
Comment (ruanxin) [inline tests/e2e/e2e_suite_test.go:38]: why remove this?

Comment (nesmabadr) [reply tests/e2e/e2e_suite_test.go:38]: We are not logging anything using this one. Also this is the controller-runtime logger

**Comment by ruanxin** [comment]
let's postpone until this PR gets merged, some of the import have to be changed from `v1beta2` to `shared`

File: tests/e2e/e2e_suite_test.go (hunk 1 above)
Comment (ruanxin) [reply tests/e2e/e2e_suite_test.go:38]: Yes, but the `envtest` is from controller-runtime, it's fine we adjust the logger setting here. But I got it, the `ConfigLogger` is coming from lifecycle-manager, it's crazy we cross-reference like this.

File: cmd/kyma/alpha/create/module/module.go [hunk 2]
```diff
@@ -468,9 +469,9 @@ func (cmd *command) getModuleTemplateAnnotations(modCnf *Config, crValidator val
 
 	isClusterScoped := isCrdClusterScoped(crValidator.GetCrd())
 	if isClusterScoped {
-		annotations[v1beta2.IsClusterScopedAnnotation] = v1beta2.EnableLabelValue
+		annotations[shared.IsClusterScopedAnnotation] = v1beta2.EnableLabelValue
 	} else {
-		annotations[v1beta2.IsClusterScopedAnnotation] = v1beta2.DisableLabelValue
+		annotations[shared.IsClusterScopedAnnotation] = v1beta2.DisableLabelValue
 	}
 	annotations["operator.kyma-project.io/module-version"] = moduleVersion
```
Comment (ruanxin) [inline cmd/kyma/alpha/create/module/module.go:476]: can you also replace with reference to shared package? please also check if there is other hardcoded values need to be referenced, feel free create followup issue if some scope becomes too big.

File: cmd/kyma/alpha/create/module/module_test.go [hunk 3]
```diff
@@ -150,7 +152,7 @@ func Test_command_getModuleTemplateAnnotations(t *testing.T) {
 				"annotation1": "value1",
 				"annotation2": "value2",
 				"operator.kyma-project.io/module-version": "1.1.1",
-				v1beta2.IsClusterScopedAnnotation:         v1beta2.DisableLabelValue,
+				shared.IsClusterScopedAnnotation:          v1beta2.DisableLabelValue,
```
Comment (ruanxin) [inline cmd/kyma/alpha/create/module/module_test.go:155]: actually this`DisableLabelValue` and `EnableLabelValue` can be also moved to `shared`, but let's handle it in different issue, can you create a followup for it? Or maybe another PR in LM.

Comment (nesmabadr) [reply cmd/kyma/alpha/create/module/module_test.go:155]: Created a PR in lifecycle manager: https://github.com/kyma-project/lifecycle-manager/pull/1159

File: cmd/kyma/alpha/list/module/module.go [hunk 4]
```diff
@@ -25,21 +23,25 @@ import (
...
+	"github.com/kyma-project/cli/internal/kube"
 )
 
 //go:embed list.tmpl
 var moduleTemplates string
 
 var moduleTemplateResource = schema.GroupVersionResource{
-	Group:    "operator.kyma-project.io",
+	Group:    shared.OperatorGroup,
 	Version:  "v1beta2",
```
Comment (ruanxin) [inline cmd/kyma/alpha/list/module/module.go:37]: this version could be `v1beta2.GroupVersion.Version`

**Review by ruanxin — APPROVED** [review]

**Pull request merged**